editor/scene_tree_editor.cpp

```diff
@@ -68,24 +68,18 @@ void SceneTreeEditor::_cell_button_pressed(Object *p_item, int p_column, int p_i
 			emit_signal("open_script", script);
 
 	} else if (p_id == BUTTON_VISIBILITY) {
-
-		if (n->is_class("Spatial")) {
-
-			bool v = bool(n->call("is_visible"));
-			undo_redo->create_action(TTR("Toggle Spatial Visible"));
-			undo_redo->add_do_method(n, "set_visible", !v);
-			undo_redo->add_undo_method(n, "set_visible", v);
-			undo_redo->commit_action();
-
-		} else if (n->is_class("CanvasItem")) {
-
-			bool v = bool(n->call("is_visible"));
-			undo_redo->create_action(TTR("Toggle CanvasItem Visible"));
-			undo_redo->add_do_method(n, v ? "hide" : "show");
-			undo_redo->add_undo_method(n, v ? "show" : "hide");
-			undo_redo->commit_action();
+		_toggle_visible(n);
+		List<Node *> selection = editor_selection->get_selected_node_list();
+		if (selection.size() > 1) {
+			for (List<Node *>::Element *E = selection.front(); E; E = E->next()) {
+				Node *nv = E->get();
+				ERR_FAIL_COND(!nv);
+				if (nv == n) {
+					continue;
+				}
+				_toggle_visible(nv);
```

Comment:
The problem with this approach is that you're creating an UndoRedo action for each step, so if you toggle visibility for 10 nodes and want to undo it, you need to press Undo 10 times.

Reply:
Ok,I get it.
I will fix it soon.